tools/scripts/tt

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+"""tt - the xfiles-dana test tool
+
+Provides functions for running emulator tests and inspecting memory
+
+Usage: tt --help
+"""
+
+import click
+from re import compile
+from os import path
+from subprocess import run, PIPE, DEVNULL, TimeoutExpired
+# from ipdb import set_trace
+from os import system
+
+
+@click.group()
+@click.option('--debug', is_flag=True)
+@click.option('--rocket_chip_project_path')
+@click.pass_context
+def tt(ctx, debug, rocket_chip_project_path):
+    ctx.obj['DEBUG'] = debug
+    rocket_chip_project_path = rocket_chip_project_path or os.environ.get(
+        'ROCKET_CHIP_PROJECT_PATH')
+    assert rocket_chip_project_path, "Must specify option rocket_chip_project_path or define ROCKET_CHIP_PROJECT_PATH environment variable"
+    ctx.obj['ROCKET_CHIP_PROJECT_PATH'] = rocket_chip_project_path
+    if debug:
+        click.echo("rocket_chip_path: {}".format(rocket_chip_project_path))
+    xfiles_dana_path = path.join(rocket_chip_project_path, 'xfiles-dana')
+    if debug:
+        click.echo("xfiles_dana_path: {}".format(xfiles_dana_path))
+    ctx.obj['XFILES_DANA_PATH'] = xfiles_dana_path
+
+
+@tt.command()
+@click.argument('test_names', nargs=-1)
+@click.pass_context
+def emu_test(ctx, test_names, timeout=300):
+    """Run a baremetal emulation test."""
+    debug = ctx.obj['DEBUG']
+    rocket_chip_path = ctx.obj['ROCKET_CHIP_PATH']
+    xfiles_dana_path = ctx.obj['XFILES_DANA_PATH']
+
+    # Build and assert paths
+    emulator_bin_path = path.join(
+        rocket_chip_path,
+        'emulator',
+        'emulator-rocketchip-VelvetConfig')
```

Comment:
The emulator name is not a known quantity and will be `emulator-[package]-[config]`. I think it's safe to just specify the name of the emulator with an additional command line option.